browser/brave_webcompat_reporter/webcompat_reporter_dialog.cc

```diff
@@ -0,0 +1,144 @@
+#include "base/values.h"
+#include "brave/common/webui_url_constants.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_window.h"
+#include "chrome/browser/ui/webui/chrome_web_contents_handler.h"
+#include "chrome/browser/ui/webui/constrained_web_dialog_ui.h"
+#include "components/guest_view/browser/guest_view_base.h"
+#include "components/web_modal/web_contents_modal_dialog_host.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/web_dialogs/web_dialog_delegate.h"
+
+using content::WebContents;
+using content::WebUIMessageHandler;
+
+constexpr int kDialogMargin = 25;
+constexpr int kDialogMinHeight = 100;
+constexpr int kDialogMaxHeight = 700;
+constexpr int kDialogMaxWidth = 375;
+
+// A ui::WebDialogDelegate that specifies the webcompat reporter's appearance.
+class WebcompatReporterDialogDelegate : public ui::WebDialogDelegate {
+ public:
+  explicit WebcompatReporterDialogDelegate(WebContents* initiator,
+      std::unique_ptr<base::DictionaryValue> params);
+  ~WebcompatReporterDialogDelegate() override;
+
+  ui::ModalType GetDialogModalType() const override;
+  base::string16 GetDialogTitle() const override;
+  GURL GetDialogContentURL() const override;
+  void GetWebUIMessageHandlers(
+      std::vector<WebUIMessageHandler*>* handlers) const override;
+  void GetDialogSize(gfx::Size* size) const override;
+  std::string GetDialogArgs() const override;
+  void OnDialogClosed(const std::string& json_retval) override;
+  void OnCloseContents(WebContents* source, bool* out_close_dialog) override;
+  bool ShouldShowDialogTitle() const override;
+
+ private:
+  WebContents* initiator_;
+  std::unique_ptr<base::DictionaryValue> params_;
+
+  DISALLOW_COPY_AND_ASSIGN(WebcompatReporterDialogDelegate);
+};
+
+WebcompatReporterDialogDelegate::WebcompatReporterDialogDelegate(WebContents* initiator,
+    std::unique_ptr<base::DictionaryValue> params)
+  : initiator_(initiator), params_(std::move(params)) {
+}
+
+WebcompatReporterDialogDelegate::~WebcompatReporterDialogDelegate() {
+}
+
+ui::ModalType WebcompatReporterDialogDelegate::GetDialogModalType() const {
+  // Not used, returning dummy value.
+  NOTREACHED();
+  return ui::MODAL_TYPE_WINDOW;
+}
+
+base::string16 WebcompatReporterDialogDelegate::GetDialogTitle() const {
+  // Only used on Windows?
+  return base::string16();
+}
+
+GURL WebcompatReporterDialogDelegate::GetDialogContentURL() const {
+  return GURL(kBraveUIWebcompatReporterURL);
+}
+
+void WebcompatReporterDialogDelegate::GetWebUIMessageHandlers(
+    std::vector<WebUIMessageHandler*>* /* handlers */) const {
+  // WebcompatReporterWebUI should add its own message handlers.
+}
+
+void WebcompatReporterDialogDelegate::GetDialogSize(gfx::Size* size) const {
+  DCHECK(size);
+
+  gfx::Size target_size;
+
+  web_modal::WebContentsModalDialogHost* host = nullptr;
+  WebContents* outermost_web_contents =
+      guest_view::GuestViewBase::GetTopLevelWebContents(initiator_);
```

Comment:
why are we using GuestViewBase here?

Reply:
I modeled that from the Tip dialog implementation [here](https://github.com/brave/brave-core/blob/df6480b37ad2b711ac442f3a249d3f71da4e5e8e/browser/brave_rewards/tip_dialog.cc#L100). Any recommendations on alternatives?

Reply:
cc @petemill because it seems like we're really overcomplicating things here. Most dialogs are very simple and a similar ConstrainedWebDialogUI (SigninEmailConfirmationDialog) only defines the width if no size is set